reana/reana_dev/run.py

```diff
@@ -604,18 +555,21 @@ def _return_missing_output_files(component: str, workflow_name: str) -> List[str
                     elif "failed" in status:
                         run_statistics["failed"].append(workflow_name)
                     elif "finished" in status or "stopped" in status:
-                        if not _verify_log_output(component, workflow_name):
-                            run_statistics["failed"].append(workflow_name)
-                            continue
-
-                        missing_files = _return_missing_output_files(
-                            component, workflow_name
-                        )
-
-                        if len(missing_files):
-                            run_statistics["failed"].append(workflow_name)
-                        else:
-                            run_statistics["passed"].append(workflow_name)
+                        try:
+                            tests = run_command(
+                                f"reana-client test -w {workflow_name}",
+                                component,
+                                return_output=True,
+                            )
+
+                            if re.findall(r".*failed$", tests, re.MULTILINE):
+                                run_statistics["failed"].append(workflow_name)
+                            else:
+                                run_statistics["passed"].append(workflow_name)
+                        except SystemExit:
```

Comment:
Mmmh catching SystemExit is (probably) not best practice, but improving this would require refactoring `run_command` which is used everywhere, so let's keep it like this for now.

However, I think that if the command fails, then we should put the workflow in the list of the failed ones?

Reply:
Otherwise this can happen:
```
Test matrix:
  - 1 demo example(s): reana-demo-root6-roofit
  - 4 workflow engine(s): cwl, serial, snakemake, yadage
  - 1 compute backend(s): kubernetes

Test results:
  - 0 submitted
  - 0 running
  - 0 pending
  - 0 queued
  - 0 passed
  - 0 failed

OK
```
Four workflows were run, but the final statistics all report 0 workflows